### PR TITLE
johndanz/ch9455/bad-position-shares-math

### PR DESCRIPTION
**Pull request by JohnDanz**
[Clubhouse Story](https://app.clubhouse.io/augur/story/9455/bad-position-shares-math)

## Submitter checklist
- [ ] Test covering functionality added/fixed in
- Check the linked story includes the following:
  - [ ] Steps to reproduce
  - [ ] Screenshot (If applicable)

## Reviewer Checklist
- [ ] Test/lint pass

### CODE REVIEW

File: src/modules/my-positions/selectors/my-positions-summary.js [hunk 1]
```diff
@@ -84,7 +84,11 @@ export const generatePositionsSummary = memoize((numPositions, qtyShares, meanTr
...
   if (typeof objects === 'number') return objects
-  if (!Array.isArray(objects) && typeof objects === 'object') return objects[property]
+  if (!Array.isArray(objects) && typeof objects === 'object') return objects
   if (!Array.isArray(objects)) return 0
-  return objects.reduce((accum, item) => item[property] + accum, 0)
+  return objects.reduce((accum, item) => {
+    const bnProperty = createBigNumber(item[property], 10)
+    const bnAccum = createBigNumber(accum, 10)
+    return bnProperty.plus(bnAccum).toString()
+  }, 0)
```
Comment (tinybike) [inline src/modules/my-positions/selectors/my-positions-summary.js]: Seems more efficient to use createBigNumber(0) as the starting value, and then not convert `accum` to bignumber for every element.

File: src/modules/my-positions/selectors/my-positions-summary.js [hunk 2]
```diff
@@ -84,7 +84,11 @@ export const generatePositionsSummary = memoize((numPositions, qtyShares, meanTr
 function accumulate(objects, property) {
   if (!objects) return 0
   if (typeof objects === 'number') return objects
-  if (!Array.isArray(objects) && typeof objects === 'object') return objects[property]
+  if (!Array.isArray(objects) && typeof objects === 'object') return objects
```
Comment (tinybike) [inline src/modules/my-positions/selectors/my-positions-summary.js]: Why's this change needed?

Comment (JohnDanz) [reply src/modules/my-positions/selectors/my-positions-summary.js]: oh, that was a mistake, Will correct.

**Comment by JohnDanz** [comment]
one more change coming in, do not merge yet.

**Pull request closed without merging**